### DISCUSSION TITLE
Adding www. to admin pages

### DISCUSSION THREAD

**Discussion by dan-it** [YOURLS/YOURLS#3182] [Plugins]
My hosting company 301-redirects https://example.com to https://www.example.com. Unfortunately I did notice only after purchasing a domain for YOURLS.

I noticed that everything seems to be fine if I set YOURLS_SITE to https://example.com except the admin pages that fail with a CORS error.

I wrote a quick and dirty plugin to insert "www." for admin pages. 

```
<?php
/*
Plugin Name: PrependWWW2adminURL
Description: Prepend "www." to admin URL
Version: 1.0
Author: dan-it
*/

yourls_add_filter( 'admin_url', 'prependWWW' );

function prependWWW( $admin, $page ) {
    if (preg_match('!^https?://www\.', $admin)) {
        return $admin;
    } else {
        return preg_replace('!^(https?://)!', '${1}www.', $admin);
    }
}
```
It works for me and may be useful to others, I'm leaving it here open to suggestions and critiques.

Ciao!
dan

**Comment by ozh** [comment]
If your hosting company enforces `www.`, either you overlooked a config somewhere, or you really should change hosting company :) 

(Seems very unlikely to me that a hosting provider would do this)

**Comment by dan-it** [comment]
He he... You're right but it's a very low budget host (tophost.it) and they offer it only for their top level plan which I do not need.

I just asked their support because of your message and they confirmed.

 Considering that the solution (plugin above) is very easy I won't bother so much and proceed with them (at least until the next renewal :D)
